From 12ca16238b8a53e45e73dc1d96c4e1b5a0601411 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Tue, 19 Mar 2019 15:49:08 +0200
Subject: [PATCH 109/235] staging: fsl-dpaa2/mac: Fix bug in pause frame
 advertising

Fix issue where advertised pause frame settings may be overwritten.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/mac/mac.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/mac/mac.c b/drivers/staging/fsl-dpaa2/mac/mac.c
index a03207e28605..eb2602e055a4 100644
--- a/drivers/staging/fsl-dpaa2/mac/mac.c
+++ b/drivers/staging/fsl-dpaa2/mac/mac.c
@@ -422,6 +422,11 @@ static void configure_link(struct dpaa2_mac_priv *priv,
 	phydev->speed = cfg->rate;
 	phydev->duplex  = !!(cfg->options & DPMAC_LINK_OPT_HALF_DUPLEX);
 
+	if (cfg->advertising != 0) {
+		phydev->advertising = 0;
+		link_mode_dpmac2phydev(cfg->advertising, &phydev->advertising);
+	}
+
 	if (phydev->supported & SUPPORTED_Pause) {
 		if (cfg->options & DPMAC_LINK_OPT_PAUSE)
 			phydev->advertising |= ADVERTISED_Pause;
@@ -436,11 +441,6 @@ static void configure_link(struct dpaa2_mac_priv *priv,
 			phydev->advertising &= ~ADVERTISED_Asym_Pause;
 	}
 
-	if (cfg->advertising != 0) {
-		phydev->advertising = 0;
-		link_mode_dpmac2phydev(cfg->advertising, &phydev->advertising);
-	}
-
 	if (cfg->options & DPMAC_LINK_OPT_AUTONEG) {
 		phydev->autoneg = AUTONEG_ENABLE;
 		phydev->advertising |= ADVERTISED_Autoneg;
-- 
2.17.1

